src/runtime_src/hip/core/context.cpp

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Device, Inc. All rights reserved.
+
+#include "common.h"
+#include "context.h"
+#include "device.h"
+
+namespace xrt::core::hip {
+context::
+context(std::shared_ptr<device> device)
+  : m_device{std::move(device)}
+{}
+
+// Global map of contexts
+xrt_core::handle_map<context_handle, std::shared_ptr<context>> context_cache;
+
+// thread local hip objects
+thread_local hip_tls_objs tls_objs;
+
+// returns current context
+// if primary context is active it is current
+// else returns top of ctx stackk
```

Comment:
```suggestion
// else returns top of ctx stack
```
It is unclear to me what is this stack about.